x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/test_pipeline/test_pipeline_flyout_tabs/tab_documents.tsx

```diff
@@ -4,98 +4,131 @@
 
 interface Props {
   validateAndTestPipeline: () => Promise<void>;
   isRunningTest: boolean;
-  isSubmitButtonDisabled: boolean;
+  form: FormHook;
 }
 
-export const DocumentsTab: React.FunctionComponent<Props> = ({
+export const DocumentsTab: FunctionComponent<Props> = ({
   validateAndTestPipeline,
-  isSubmitButtonDisabled,
   isRunningTest,
+  form,
 }) => {
   const { services } = useKibana();
 
+  const [, formatData] = useFormData({ form });
+
+  const onAddDocumentHandler = useCallback(
+    (document) => {
+      const { documents: existingDocuments } = formatData();
+
+      form.reset({ defaultValue: { documents: [...existingDocuments, document] } });
+    },
+    [form, formatData]
+  );
+
   return (
-    <div data-test-subj="documentsTabContent">
-      <EuiText>
-        <p>
-          <FormattedMessage
-            id="xpack.ingestPipelines.testPipelineFlyout.documentsTab.tabDescriptionText"
-            defaultMessage="Provide an array of documents for the pipeline to ingest. {learnMoreLink}"
-            values={{
-              learnMoreLink: (
-                <EuiLink
-                  href={`${services.documentation.getEsDocsBasePath()}/simulate-pipeline-api.html`}
-                  target="_blank"
-                  external
-                >
-                  {i18n.translate(
-                    'xpack.ingestPipelines.testPipelineFlyout.documentsTab.simulateDocumentionLink',
-                    {
-                      defaultMessage: 'Learn more.',
-                    }
-                  )}
-                </EuiLink>
+    <Form
+      form={form}
+      data-test-subj="testPipelineForm"
+      isInvalid={form.isSubmitted && !form.isValid}
+      onSubmit={validateAndTestPipeline}
+      error={form.getErrors()}
+    >
+      <div data-test-subj="documentsTabContent">
+        <EuiText>
+          <p>
+            <FormattedMessage
+              id="xpack.ingestPipelines.testPipelineFlyout.documentsTab.tabDescriptionText"
+              defaultMessage="Provide documents for the pipeline to ingest. {learnMoreLink}"
+              values={{
+                learnMoreLink: (
+                  <EuiLink
+                    href={`${services.documentation.getEsDocsBasePath()}/simulate-pipeline-api.html`}
+                    target="_blank"
+                    external
+                  >
+                    {i18n.translate(
+                      'xpack.ingestPipelines.testPipelineFlyout.documentsTab.simulateDocumentionLink',
+                      {
+                        defaultMessage: 'Learn more.',
+                      }
+                    )}
+                  </EuiLink>
+                ),
+              }}
+            />
+          </p>
+        </EuiText>
+
+        <EuiSpacer size="m" />
+
+        <ImportDocumentsAccordion onAddDocuments={onAddDocumentHandler} />
+
+        <EuiSpacer size="l" />
+
+        {/* Documents editor */}
+        <UseField
```

Comment:
WDYT of surfacing the number of documents somewhere? For example in the label:

![image](https://user-images.githubusercontent.com/1238659/93932503-29ef3e80-fcd5-11ea-8a51-3baffa6f1191.png)

Reply:
I like this idea. It would require some refactoring though, since the field label is currently defined in the form schema. I'm going to leave as-is for now.